### PR TITLE
Add insert ignore

### PR DESCRIPTION
**Pull request by samsonasik**
- [x] Are you creating a new feature?
  - [x] Why is the new feature needed? What purpose does it serve?
            Useful to ignore duplicate entry error.
  - [x] How will users use the new feature?

```php
$insert= new InsertIgnore('tablename');
$insert->values($data);

$statement = $sql->prepareStatementForSqlObject($insert);
$statement->execute();
```
  - [x] Base your feature on the `develop` branch, and submit against that branch.
  - [x] Add only one feature per pull request; split multiple features over multiple pull requests
  - [x] Add tests for the new feature.
  - [x] Add a `CHANGELOG.md` entry for the new feature.

### CODE REVIEW

File: src/Sql/InsertIgnore.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Zend\Db\Sql;
+
+class InsertIgnore extends Insert
+{
+    /**
+     * @var array Specification array
+     */
+    protected $specifications = [
+        self::SPECIFICATION_INSERT => 'INSERT IGNORE INTO %1$s (%2$s) VALUES (%3$s)',
```
Comment (Ocramius) [inline src/Sql/InsertIgnore.php]: Are all supported drivers capable of evaluating `INSERT IGNORE`?

File: src/Sql/InsertIgnore.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Db\Sql;
+
+class InsertIgnore extends Insert
+{
+    /**
+     * @var array Specification array
```
Comment (Ocramius) [inline src/Sql/InsertIgnore.php]: Please specify the type of the array

File: test/unit/Sql/InsertIgnoreTest.php [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+     * Sets up the fixture, for example, opens a network connection.
+     * This method is called before a test is executed.
+     */
+    protected function setUp()
+    {
+        $this->insert = new InsertIgnore;
+    }
+
+    /**
+     * @covers \Zend\Db\Sql\InsertIgnore::into
```
Comment (Ocramius) [inline test/unit/Sql/InsertIgnoreTest.php]: I suggest dropping all these `@covers`, ans only adding one at class level

File: src/Sql/InsertIgnore.php (hunk 1 above)
Comment (samsonasik) [reply src/Sql/InsertIgnore.php]: probably MySQL only, that's why I didn't add `insertIgnore()` function to `Sql` and `AbstractTableGateway`

File: src/Sql/InsertIgnore.php (hunk 2 above)
Comment (samsonasik) [reply src/Sql/InsertIgnore.php]: it is copied originally from `Insert` class, do you have sample of "type of the array"?

File: test/unit/Sql/InsertIgnoreTest.php (hunk 3 above)
Comment (samsonasik) [reply test/unit/Sql/InsertIgnoreTest.php]: removed `@covers`

File: src/Sql/InsertIgnore.php (hunk 2 above)
Comment (Ocramius) [reply src/Sql/InsertIgnore.php]: `string[]` in this case, or even `array<string, string>` (unsure if the tooling craps out with that, though)

File: src/Sql/InsertIgnore.php (hunk 1 above)
Comment (Ocramius) [reply src/Sql/InsertIgnore.php]: Hmm, can we please survey that before introducing it then?

**Comment by michalbundyra** [comment]
Thanks, @samsonasik!

**Pull request merged**